assets/src/edit-story/components/keyboard/index.js

```diff
@@ -0,0 +1,183 @@
+/*
+ * Copyright 2020 Google LLC
+ *
+ * Licensed under the Apache License, Version 2.0 (the "License");
+ * you may not use this file except in compliance with the License.
+ * You may obtain a copy of the License at
+ *
+ *     https://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+
+/**
+ * External dependencies
+ */
+import Mousetrap from 'mousetrap';
+
+/**
+ * WordPress dependencies
+ */
+import { useContext, useEffect, createRef } from '@wordpress/element';
+
+/**
+ * Internal dependencies
+ */
+import Context from './context';
+
+const PROP = '__WEB_STORIES_MT__';
+const NON_EDITABLE_INPUT_TYPES = ['submit', 'button', 'checkbox', 'radio'];
+
+const globalRef = createRef();
+
+/**
+ * See https://craig.is/killing/mice#keys for the supported key codes.
+ *
+ * @param {{current: Node}} ref
+ * @param {string|Array|Object} keyNameOrSpec
+ * @param {function(KeyboardEvent)} callback
+ * @param {Array|undefined} deps
+ */
+export function useKeyDownEffect(
+  ref,
+  keyNameOrSpec,
+  callback,
+  deps = undefined
+) {
+  const { keys } = useContext(Context);
```

Comment:
Why the context here? There's no provider for it yet, right? But it's for future localization of the shortcuts?

Reply:
Yeah. A bit of the future proofing, since it doesn't cost really anything. I can remove for now if you'd prefer. TBH, it's most important for customization or any multiple short-cut cases. Otherwise, localization is fairly simple inline now. E.g.

```
useKeyDownEffect(ref, __('ctrl+b', 'web-stories'), ...);
```